### PR TITLE
Lut inferable quantizer

### PR DESCRIPTION
**Pull request by lapid92**
* Add Keras LUT inferable quantizers

### CODE REVIEW

File: model_compression_toolkit/quantizers_infrastructure/keras/inferable_quantizers/activation_inferable_quantizers/activation_lut_pot_inferable_quantizer.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            Args:
+                inputs: input tensor to quantize
+
+            Returns:
+                quantized tensor.
+            """
+            assert inputs.dtype == tf.float32, f'Input tensor was expected to be a float tensor but is of type ' \
+                                               f'{inputs.dtype}'
+
+            return lut_kmeans_quantizer(inputs, cluster_centers=self.cluster_centers, signed=True,
```
Comment (elad-c) [inline model_compression_toolkit/quantizers_infrastructure/keras/inferable_quantizers/activation_inferable_quantizers/activation_lut_pot_inferable_quantizer.py:90]: why must the quantization be signed? Should use self.signed

File: model_compression_toolkit/quantizers_infrastructure/keras/inferable_quantizers/activation_inferable_quantizers/activation_lut_pot_inferable_quantizer.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    import tensorflow as tf
+    from model_compression_toolkit.quantizers_infrastructure.keras.inferable_quantizers.base_keras_inferable_quantizer \
+        import BaseKerasInferableQuantizer
+    from model_compression_toolkit.quantizers_infrastructure.keras.quantizer_utils import lut_kmeans_quantizer
+
+
+    @mark_quantizer(quantization_target=QuantizationTarget.Activation,
+                    quantization_method=[QuantizationMethod.LUT_POT_QUANTIZER],
+                    quantizer_type=None)
+    class ActivationLutPOTInferableQuantizer(BaseKerasInferableQuantizer):
```
Comment (elad-c) [inline model_compression_toolkit/quantizers_infrastructure/keras/inferable_quantizers/activation_inferable_quantizers/activation_lut_pot_inferable_quantizer.py:36]: what about ActivationLutSymmetricInferableQuantizer

File: model_compression_toolkit/quantizers_infrastructure/keras/quantizer_utils.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+# limitations under the License.
+# ==============================================================================
+
+import numpy as np
+import tensorflow as tf
+
+from model_compression_toolkit.quantizers_infrastructure.common.constants import EPS, MULTIPLIER_N_BITS
+
+
+def lut_kmeans_quantizer(tensor_data: tf.Tensor,
```
Comment (elad-c) [inline model_compression_toolkit/quantizers_infrastructure/keras/quantizer_utils.py:22]: if this function quantizes a tensor according to pre-defined cluster centers, then kmeans isn't relevant here.

File: tests/quantizers_infrastructure_tests/keras_tests/test_inferable_quantizers/test_activation_inferable_quantizers.py [hunk 4]
```diff
@@ -287,3 +288,81 @@ def test_uniform_inferable_quantizer_zero_not_in_range(self):
...
+        self.assertTrue(quantizer_config['num_bits'] == num_bits)
+        self.assertTrue(np.all(quantizer_config['cluster_centers'] == cluster_centers))
+        self.assertTrue(quantizer_config['threshold'] == thresholds)
+        self.assertTrue(quantizer_config['signed'] == signed)
+
+        # Initialize a random input to quantize between -50 to 50.
+        input_tensor = tf.constant(np.random.rand(1, 50, 50, 3) * 100 - 50, tf.float32)
+        fake_quantized_tensor = quantizer(input_tensor)
+
+        quant_tensor_values = tf.concat([(center / (2 ** (MULTIPLIER_N_BITS - 1))) * thresholds for
```
Comment (elad-c) [inline tests/quantizers_infrastructure_tests/keras_tests/test_inferable_quantizers/test_activation_inferable_quantizers.py:326]: use np.concatenate, as it's all numpy arrays

**Review by elad-c — CHANGES_REQUESTED** [review]

File: model_compression_toolkit/quantizers_infrastructure/keras/inferable_quantizers/weights_inferable_quantizers/weights_lut_symmetric_inferable_quantizer.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+            self.num_bits = num_bits
+            self.cluster_centers = cluster_centers
+
+            self.per_channel = per_channel
+            self.channel_axis = channel_axis
+            self.input_rank = input_rank
+
+            # Tensorflow's fake_quant_with_min_max_vars_per_channel only works on last axis, so
+            # need to move the quantization axis to the last axis
```
Comment (elad-c) [inline model_compression_toolkit/quantizers_infrastructure/keras/inferable_quantizers/weights_inferable_quantizers/weights_lut_symmetric_inferable_quantizer.py:85]: Do we actually need to move the channels to the last axis? we're not using TF fake_quant function. Can remove input_rank and all the permutataion if that's true

**Review by elad-c — APPROVED** [review]

**Review by elad-c — APPROVED** [review]

**Review by elad-c — APPROVED** [review]
